src/analysis/analytic/basic.lean

```diff
@@ -448,3 +453,315 @@ begin
   { simp at h,
     simp [h, continuous_on_empty] }
 end
+
+end
+
+/-!
+### Changing origin in a power series
+
+If a function is analytic in a disk `D(x, R)`, then it is analytic in any disk contained in that
+one. Indeed, one can write
+$$
+f (x + y + z) = \sum_{n} p_n (y + z)^n = \sum_{n, k} \choose n k p_n y^{n-k} z^k
+= \sum_{k} (\sum_{n} \choose n k p_n y^{n-k}) z^k.
+$$
+The corresponding power series has thus a `k`-th coefficient equal to
+`\sum_{n} \choose n k p_n y^{n-k}`. In the general case where `pₙ` is a multilinear map, this has
+to be interpreted suitably: instead of having a binomial coefficient, one should sum over all
+possible subsets `s` of `fin n` of cardinal `k`, and attribute `z` to the indices in `s` and
+`y` to the indices outside of `s`.
+
+In this paragraph, we implement this. The new power series is called `p.change_origin y`. Then, we
+check its convergence and the fact that its sum coincides with the original sum. The outcome of this
+discussion is that the set of points where a function is analytic is open.
+-/
+
+namespace formal_multilinear_series
+
+variables (p : formal_multilinear_series 𝕜 E F) {x y : E} {r : nnreal}
+
+/-- Changing the origin of a formal multilinear series `p`, so that
+`p.sum (x+y) = (p.change_origin x).sum y` when this makes sense. We don't use the bracket notation
+`⟨n, s, hs⟩` as it leads to a bad definition, but we will try to use it as much as possible in
+the proofs below to increase readability. -/
+def change_origin (x : E) :
+  formal_multilinear_series 𝕜 E F :=
+λ k, tsum (λi, (p i.1).restr i.2.1 i.2.2 x :
+  (Σ (n : ℕ), {s : finset (fin n) // finset.card s = k}) → (E [×k]→L[𝕜] F))
+
+/-- Auxiliary lemma controlling the summability of the sequence appearing in the definition of
+`p.change_origin`, first version. -/
+lemma change_origin_summable_aux1 (h : (nnnorm x + r : ennreal) < p.radius) :
```

Comment:
```suggestion
-- Note here and below it is necessary to use `@` and provide implicit arguments using `_`,
-- so that it is possible to use pattern matching in the lambda.
-- Overall this seems a good trade-off in readability.
lemma change_origin_summable_aux1 (h : (nnnorm x + r : ennreal) < p.radius) :
```